### PR TITLE
SLING-12350 - Make the build work on more recent JDKs

### PR DESCRIPTION
**Pull request by csaboka**
- Update parent POM version
- Update Spotbugs plugin to a version that runs on recent JDKs, then suppressed the newly activated rules to keep the build working.
- Remove unit tests that relied on modifying Java system classes via reflection. The mocked static method wasn't called for anything related to the code under test, so this change seems to be harmless.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <parent>
         <groupId>org.apache.sling</groupId>
         <artifactId>sling-bundle-parent</artifactId>
-        <version>44</version>
+        <version>52</version>
```
Comment (kwin) [inline pom.xml:28]: Can you update to latest? There are hints on what you need to consider in https://cwiki.apache.org/confluence/x/SI75E

Comment (csaboka) [reply pom.xml:28]: The latest parent only builds with Java 17 or later. Is that OK? Aren't there any clients that need to run on JDK 11?

Comment (csaboka) [reply pom.xml:28]: Oh, after reading the linked page I can see that it still targets JDK 11, but requires JDK 17+ for the build process. I'll have a try at it.

**Comment by csaboka** [comment]
Well, now I get the blame for all the code duplication and missing coverage on the code I "touched". I hope that isn't a blocker for merging this.

File: pom.xml [hunk 2]
```diff
@@ -41,27 +41,133 @@
...
 
     <scm>
         <connection>scm:git:https://gitbox.apache.org/repos/asf/sling-org-apache-sling-scripting-sightly-compiler.git</connection>
         <developerConnection>scm:git:https://gitbox.apache.org/repos/asf/sling-org-apache-sling-scripting-sightly-compiler.git</developerConnection>
-        <url>https://github.com/apache/sling-org-apache-sling-scripting-sightly-compiler.git</url>
         <tag>org.apache.sling.scripting.sightly.compiler-1.2.6-1.4.0</tag>
+        <url>https://github.com/apache/sling-org-apache-sling-scripting-sightly-compiler.git</url>
     </scm>
 
     <properties>
```
Comment (kwin) [inline pom.xml:54]: I would suggest setting an explicit java bytecode version (no need to upgrade > 8 for now)

**Comment by csaboka** [comment]
@kwin , can you merge this? Is there anything still missing?

**Comment by kwin** [comment]
@csaboka Can you rebase to only have 3 commits (parent update, reformat, .git-blame-ignore-revs) and make sure that all commit messages have the prefix of the related JIRA ticket? Thanks.

**Comment by csaboka** [comment]
@kwin : Whoops, it didn't have a related JIRA ticket, but I have created one now. I've also rebased the code as requested. Is it good to merge now?

**Pull request merged**